consensus/replay_test.go

```diff
@@ -104,3 +150,68 @@ func openWAL(t *testing.T, cs *ConsensusState, file string) {
 	wal.exists = true
 	cs.wal = wal
 }
+
+//-----------------------------------------------
+// Test the log at every iteration, and set the privVal last step
+// as if the log was written after signing, before the crash
+
+func TestReplayCrashAfterWrite(t *testing.T) {
+	split := strings.Split(testLog, "\n")
+	for i := 0; i < len(split)-1; i++ {
+		cs, newBlockCh, _, f := setupReplayTest(i+1, true)
+		runReplayTest(t, cs, f, newBlockCh)
```

Comment:
runReplayTest expects a new block.  Is it expected that all WAL prefixes (e.g. the first i lines for i := 0; i < len(split)) will generate new blocks?  Can we add a comment explaining why?  Cuz it isn't obvious to me.